projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -246,18 +248,49 @@ public function add_stats_menu() {
 		add_menu_page( __( 'Stats', 'jetpack' ), $menu_title, 'edit_posts', 'https://wordpress.com/stats/day/' . $this->domain, null, 'dashicons-chart-bar', 3 );
 	}
 
+	/**
+	 * Holds the current product set by get_current_product().
+	 *
+	 * @var string
+	 */
+	private $cached_product = null;
```

Comment:
It's kinda sorta cached. I named this "cached_product" because I didn't want someone to reference $this->product without knowing how it was set.